sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -769,10 +779,38 @@ void CAFMaker::beginSubRun(art::SubRun& sr) {
   if(bnb_spill){
     FillExposure(*bnb_spill, fBNBInfo, fSubRunPOT);
     fTotalPOT += fSubRunPOT;
+
+    // Find the spill for each event and fill the event map:
+    // We take the latest spill for a given event number to be the one to keep
+    fHasBNBInfo = true;
+    for(const sbn::BNBSpillInfo& info: *bnb_spill)
+    {
+      if ( fBNBInfoEventMap.find(info.event)==fBNBInfoEventMap.end() ) {
+        fBNBInfoEventMap[info.event] = info;
+      }
+      else if ( (info.spill_time_s+(info.spill_time_ns/1.0e9)) >
```

Comment:
Can you put this into a function, like:
```cpp
namespace {
  template <typename SpillInfo>
  double spillInfoToTimestamp(SpillInfo const& info) {
    return static_cast<double>(info.spill_time_s)
      + static_cast<double>(info.spill_time_ns)*1e-9;
  }
}
```
Even better would be to add a `timestamp()` member function to `caf::SRBNBSpillInfo` and sister.

Reply:
Good idea! Is the `namespace { }` enclosure here just a generic template of such a function or do you prefer it live in a specific namespace? Otherwise, I was thinking to just put it in `caf` namespace in the code.

Reply:
The [unnamed namespace](https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces) has a specific meaning, in that its content is not visible outside the ~translation unit~ C++ source file where it is defined. For example, if I wrote a header file `cheater.h` like:
```cpp
template <typename SpillInfo>
double spillInfoToTimestamp(SpillInfo const& info);
```
and I included it in another source file:
```cpp
#include "cheater.h"

double const timeStamp = spillInfoToTimestamp(SpillInfo{});
```
this would fail linking because `spillInfoToTimestamp` would not be found anywhere.

[example faults: (1) for a template this should not be done in such way anyway (pretend it's not a template) and (2) the header where `SpillInfo` is declared needs to be included somewhere]